cmake/thirdparty/BLTSetupCUDA.cmake

```diff
@@ -155,21 +155,24 @@ endif()
 # leaving it to the default source file language.
 # This logic is handled in the blt_add_library/executable
 # macros
-blt_import_library(NAME          cuda
-                   COMPILE_FLAGS ${_cuda_compile_flags}
-                   INCLUDES      ${CUDA_INCLUDE_DIRS}
-                   LIBRARIES     ${CUDA_LIBRARIES}
-                   LINK_FLAGS    "${CMAKE_CUDA_LINK_FLAGS}"
-                   EXPORTABLE    ${BLT_EXPORT_THIRDPARTY}
-                   )
+if (NOT TARGET cuda)
```

Comment:
I think we should move this check at the very beginning of the file or at the BLTSetupTargets.cmake file level. As stated before by @cyrush  we want to skip all logic here if already run.

Either:

```
if (TARGET cuda)
  return()
endif()
```

or:

```
if (NOT TARGET cuda)
  message(STATUS "CUDA Support is ${BLT_ENABLE_CUDA}")
  if (BLT_ENABLE_CUDA AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/thirdparty/BLTSetupCUDA.cmake")
    include("${CMAKE_CURRENT_LIST_DIR}/thirdparty/BLTSetupCUDA.cmake")
  endif()
endif()
```